state/mysql/mysql.go

```diff
@@ -231,7 +267,35 @@ func (m *MySQL) finishInit(ctx context.Context, db *sql.DB) error {
 	}
 
 	// will be nil if everything is good or an err that needs to be returned
-	return m.ensureStateTable(ctx, m.tableName)
+	if err = m.ensureStateTable(ctx, m.schemaName, m.tableName); err != nil {
+		return err
+	}
+
+	if err = m.ensureMetadataTable(ctx, m.schemaName, m.metadataTableName); err != nil {
+		return err
+	}
+
+	if m.cleanupInterval != nil {
+		gc, err := sqlCleanup.ScheduleGarbageCollector(sqlCleanup.GCOptions{
```

Comment:
To my understanding, this may not work with how the GC works.

Per a comment here: https://stackoverflow.com/a/16588699

> This is leaving the fields the same, but it's still doing an update either way.

The problem is that the GC library expects UpdateLastCleanupQuery to return `rowsAffected == 0` if the update isn't needed. According to docs, this one always updates the database, so I presume that `rowsAffected == 1`, which would cause the cleanup to run every time.

Perhaps this should use a stored procedure instead? Or something else

Reply:
Yup, the two different logs above are for mysql and maridb.
Just, for re-confirming, I added connection string as well to the log:
For mysql:
```
{"Time":"2023-03-23T06:59:23.453716692Z","Action":"output","Package":"github.com/dapr/components-contrib/tests/certification/state/mysql","Test":"TestMySQL/cleanup/cleanup_concurrency","Output":"time=\"2023-03-23T06:59:23.453657992Z\" level=info msg=\"Testing CleanupExpired - no rows should be affected, %sroot:root@tcp(localhost:3306)/?allowNativePasswords=true\" instance=fv-az365-158 scope=dapr.components type=log ver=unknown\n"}
{"Time":"2023-03-23T06:59:23.454251799Z","Action":"output","Package":"github.com/dapr/components-contrib/tests/certification/state/mysql","Test":"TestMySQL/cleanup/cleanup_concurrency","Output":"time=\"2023-03-23T06:59:23.454176798Z\" level=info msg=\"=========================Rows affected: 0\" instance=fv-az365-158 scope=dapr.components type=log ver=unknown\n"}
```
For mariadb:
```
{"Time":"2023-03-23T06:59:31.507769379Z","Action":"output","Package":"github.com/dapr/components-contrib/tests/certification/state/mysql","Test":"TestMySQL/cleanup#01/cleanup_concurrency","Output":"time=\"2023-03-23T06:59:31.507718579Z\" level=info msg=\"Testing CleanupExpired - no rows should be affected, %sroot:root@tcp(localhost:3307)/\" instance=fv-az365-158 scope=dapr.components type=log ver=unknown\n"}
{"Time":"2023-03-23T06:59:31.508253385Z","Action":"output","Package":"github.com/dapr/components-contrib/tests/certification/state/mysql","Test":"TestMySQL/cleanup#01/cleanup_concurrency","Output":"time=\"2023-03-23T06:59:31.508174984Z\" level=info msg=\"=========================Rows affected: 0\" instance=fv-az365-158 scope=dapr.components type=log ver=unknown\n"}
```